File: src/main/java/me/bdx/nhplugin/files/DataQueue.java
package me.bdx.nhplugin.files;

import java.util.ArrayList;
import java.util.HashMap;

public class DataQueue {

    private HashMap<String, ArrayList<String>> joinQueue;
    private HashMap<String, String> generalDataQueue;

    public DataQueue(){
        joinQueue = new HashMap<>();
        generalDataQueue = new HashMap<>();
    }

    /**
     * Gets the current join data queue hashmap
     * @return hashmap
     */
    public HashMap<String, ArrayList<String>> getJoinQueue(){
        return joinQueue;
    }

    /**
     * Adds the key-value pair to the join queue
     * @param key String
     * @param value ArrayList
     */
    public void addJoinQueue(String key, ArrayList<String> value){
        joinQueue.put(key, value);
    }

    /**
     * Removes the given key-value from the join queue
     * @param key String
     * @param value ArrayList
     */
    public void removeJoinQueue(String key, ArrayList<String> value){
        joinQueue.remove(key, value);
    }

    /**
     * Removes the given key from the hashmap
     * @param key String
     */
    public void removeJoinQueue(String key){
        joinQueue.remove(key);
    }

    /**
     * Gets the general data queue
     * @return The hashmap of the general data queue
     */
    public HashMap<String, String> getGeneralDataQueue() {
        return generalDataQueue;
    }

    /**
     * Sets the general data queue to the given hashmap
     * @param generalDataQueue HashMap of String and String
     */
    public void setGeneralDataQueue(HashMap<String, String> generalDataQueue) {
        this.generalDataQueue = generalDataQueue;
    }

    /**
     * Adds the given key-value pair to the general data queue
     * @param key String
     * @param value String
     */
    public void addGeneralQueue(String key, String value){
        generalDataQueue.put(key,value);
    }
}

File: example-scripts/multi-script example/scripts/utils/dataQueue.js
/** Adds the given key(playerName) and Value to the joinQueue
 *  @param key The playerName
 *  @param value The Arraylist of what is to be performed when the player joins
 */
 function addJoinQueue(key, value, nhplugin){

    var a = new ArrayList()
    for(var i=0; i<value.length; i++){
        a.add(value[i])
    }
    nhplugin.getDataQueue().addJoinQueue(key, a)
}

/** Gets the join queue
 *  @param nhplugin the Nhplugin instance
 *  @return joinQueue The ArrayList join queue
*/
function getJoinQueue(nhplugin){
    return nhplugin.getDataQueue().getJoinQueue()
}

/** Checks if the given player is in the data queue
 *  @param player The name of the player being checked for
 *  @param nhplugin The Nhplugin instance
 *  @return boolean Whether or not the player is present in the joinQueue
 */
function checkIfPlayerInQueue(player, nhplugin){
    var queue = getJoinQueue(nhplugin);
    return queue.containsKey(player)
}

/** Gets the queue data for the given playerName. Assumes that playerName is a valid data key
 *  @param player The name of the player
 *  @param nhplugin The Nhplugin instance
 *  @return The queue data for the player
 */
function getQueueData(player, nhplugin){
    var queue = getJoinQueue(nhplugin);

    return queue.get(player)
}

/** Removes the given player from the joinQueue
 *  @param playerName The name of the player being removed from the Queue
 *  @param nhplugin The instance of the Nhplugin
 */
function removeFromJoinQueue(playerName, nhplugin){
    nhplugin.getDataQueue().removeJoinQueue(playerName)
}